pkg/lib/cockpit.d.ts

```diff
@@ -368,4 +368,18 @@ declare module 'cockpit' {
 
     export let localStorage: Storage;
     export let sessionStorage: Storage;
+
+    export interface HttpOptions {
+        address: string;
+        port: number;
+        headers: { [key: string]: string };
+      }
+    
+      export interface HttpInstance {
+        request(options: { path: string; method: string; body: string }): Promise<string>;
+        get(path: string): Promise<string>;
+        post(path: string, body?: any): Promise<string>;
```

Comment:
The typechecker also [complains about this](https://github.com/cockpit-project/cockpit/actions/runs/13713605240/job/38356564208?pr=21678#step:2:472):

> Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

The docs give the allowed types:

> body can be a string, or a javascript plain object, which will be encoded as JSON data. If body is undefined or null then an empty HTTP body will be sent.

Reply:
@martinpitt thank _you_ for the review! I'm brand-new to typescript, and am subsisting on a bit of documentation and a lot of GPT. But this will be a good opportunity to understand better.

I'll try to update the typing and submit an updated PR.